packages/api/src/tasks/checkAlerts.ts

```diff
@@ -160,27 +161,134 @@ export const notifyChannel = async ({
             }),
       });
 
-      // ONLY SUPPORTS SLACK WEBHOOKS FOR NOW
       if (webhook?.service === 'slack') {
-        await slack.postMessageToWebhook(webhook.url, {
-          text: message.title,
-          blocks: [
-            {
-              type: 'section',
-              text: {
-                type: 'mrkdwn',
-                text: `*<${message.hdxLink} | ${message.title}>*\n${message.body}`,
-              },
-            },
-          ],
-        });
+        await handleSendSlackWebhook(webhook, message);
+      } else if (webhook?.service === 'generic') {
+        await handleSendGenericWebhook(webhook, message);
       }
       break;
     }
     default:
       throw new Error(`Unsupported channel type: ${channel}`);
   }
 };
+
+const handleSendSlackWebhook = async (
+  webhook: IWebhook,
+  message: {
+    hdxLink: string;
+    title: string;
+    body: string;
+  },
+) => {
+  await slack.postMessageToWebhook(webhook.url, {
+    text: message.title,
+    blocks: [
+      {
+        type: 'section',
+        text: {
+          type: 'mrkdwn',
+          text: `*<${message.hdxLink} | ${message.title}>*\n${message.body}`,
+        },
+      },
+    ],
+  });
+};
+
+export const handleSendGenericWebhook = async (
+  webhook: IWebhook,
+  message: {
+    hdxLink: string;
+    title: string;
+    body: string;
+  },
+) => {
+  // QUERY PARAMS
+  let url: string;
+  // user input of queryParams is disabled on the frontend for now
+  if (webhook.queryParams) {
+    const queryParams = JSON.parse(webhook.queryParams);
+    const queryParamsString = new URLSearchParams(queryParams).toString();
+
+    // user may have included params in both the url and the query params
+    // so they should be merged
+    const tmpURL = new URL(webhook.url);
+    if (queryParamsString) {
+      tmpURL.search = tmpURL.search
+        ? `${tmpURL.search}&${queryParamsString}`
+        : queryParamsString;
+    }
+    url = tmpURL.toString();
+  } else {
+    // if there are no query params given, just use the url
+    url = webhook.url;
+  }
+
+  const parsedHeaders = webhook.headers ? JSON.parse(webhook.headers) : {};
+
+  // HEADERS
+  // TODO: handle real webhook security and signage after v0
+  // X-HyperDX-Signature FROM PRIVATE SHA-256 HMAC, time based nonces, caching functionality etc
+  const headers = {
+    'Content-Type': 'application/json', // default, will be overwritten if user has set otherwise
+    ...parsedHeaders,
+  };
+
+  // BODY
+  let parsedBody: Record<string, string | number | symbol> = {};
+  if (webhook.body) {
+    const injectedBody = injectIntoPlaceholders(webhook.body, {
+      $HDX_ALERT_URL: message.hdxLink,
+      $HDX_ALERT_TITLE: message.title,
+      $HDX_ALERT_BODY: message.body,
+    });
+    parsedBody = JSON.parse(injectedBody);
+  }
+
+  try {
+    const response = await fetch(url, {
+      method: 'POST',
+      headers: headers,
+      body: JSON.stringify(parsedBody),
+    });
```

Comment:
nit: I think we'd need a request-error-tolerant api client here (handling retry for example). Don't need to do it in this PR and we can add it later

Reply:
yeah I think thats very fair, ill leave a comment for posterity so it can go in a v1+ addition